source/common/router/config_impl.cc

```diff
@@ -383,6 +383,41 @@ RouteConstSharedPtr PathRouteEntryImpl::matches(const Http::HeaderMap& headers,
   return nullptr;
 }
 
+RegexRouteEntryImpl::RegexRouteEntryImpl(const VirtualHostImpl& vhost,
+                                         const envoy::api::v2::Route& route,
+                                         Runtime::Loader& loader)
+    : RouteEntryImplBase(vhost, route, loader),
+      regex_(std::regex{route.match().regex(), std::regex::optimize}) {}
+
+void RegexRouteEntryImpl::finalizeRequestHeaders(Http::HeaderMap& headers) const {
+  RouteEntryImplBase::finalizeRequestHeaders(headers);
+
+  const Http::HeaderString& path = headers.Path()->value();
+  const char* query_string_start = strchr(path.c_str(), '?');
+  if (query_string_start == nullptr) {
+    query_string_start = path.c_str() + path.size();
```

Comment:
Why exclude query string?

Reply:
This was specified in the [rds API](https://github.com/lyft/envoy-api/blob/master/api/rds.proto), line 78